cpp/src/io/shp/polygon_shp_reader.cu

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <math.h>
+#include <cuda_runtime.h>
+#include <thrust/device_vector.h>
+#include <rmm/thrust_rmm_allocator.h>
+#include <cudf/types.h>
+#include <cudf/legacy/column.hpp>
+#include <utilities/error_utils.hpp>
+#include <cuspatial/shp_readers.hpp>
+#include <utility/utility.hpp>
+
+#include <ogrsf_frmts.h>
+
+using namespace std;
+
+namespace cuspatial
+{
+ 
```

Comment:
Only public APIs should be in namespace `cuspatial`. Functions only used inside this file should be in an anonymous namespace. Functions used within cuspatial across files should be in `cuspatial::detail`.

Reply:
Got it. Will change.

Reply:
Done. One function (with a new name polygon_from_shapefile ) is left inside cuspatial namespace as it uses struct polygons<T> which is defined inside cuspatial namespace.

Reply:
changed.